tests/backup/backup_resiliency_test.go

```diff
@@ -1312,10 +1312,12 @@ var _ = Describe("{RebootNodesWhenBackupsAreInProgress}", func() {
 		log.Infof("Deleting the deployed applications on application cluster")
 		opts := make(map[string]bool)
 		opts[SkipClusterScopedObjects] = true
-		ValidateAndDestroy(contexts, opts)
+		DestroyApps(contexts, opts)
 		log.Infof("Switching cluster context back to source cluster")
-		err = SetSourceKubeConfig()
-		log.FailOnError(err, "Switching context to source cluster")
+		defer func() {
+			err = SetSourceKubeConfig()
+			log.FailOnError(err, "Switching context to source cluster")
+		}()
```

Comment:
Correct me if my understanding is wrong
Here if we use defer , it will not run that block of SetSourceKubeConfig , go to 
ctx, err := backup.GetAdminCtxFromSecret() and fail as context was of destination cluster
Adding err = SetSourceKubeConfig() in defer is correct but we need to add err = SetSourceKubeConfig() again before line 1321 i.e getting the src cluster context

Reply:
@mkoppal-px Please confirm

Reply:
@kshithijiyer-px we need to keep the existing context switch to source cluster in JustAfterEach l. Just make sure you add this defer block to the first line of JustAfterEach

Reply:
Ack doing it.